mcs_benchmark_data/cli/cli.py

```diff
@@ -12,19 +13,50 @@
 
 
 class Cli:
+    __COMMAND_CLASSES = {
+        "create-benchmark-pipeline": CreateBenchmarkPipelineCommand,
+        "create-submission-pipeline": CreateSubmissionPipelineCommand,
+    }
+
     def __init__(self):
-        self.__commands = {
-            "create-benchmark-pipeline": CreateBenchmarkPipelineCommand(),
-            "create-submission-pipeline": CreateSubmissionPipelineCommand(),
-        }
+        self.__arg_parser = ArgParser()
+        self.__logger = logging.getLogger(self.__class__.__name__)
 
-    @staticmethod
-    def __add_global_args(arg_parser: ArgParser):
-        arg_parser.add_argument(
-            "--logging-level",
-            help="set logging-level level (see Python logging module)",
-        )
-        arg_parser.add_argument("-f", "--force", action="store_true")
+    def __add_arguments(self):
+        arg_parsers = [self.__arg_parser]
+​
+        subparsers = self.__arg_parser.add_subparsers(dest="command")
+        for command_name, command_class in self.__COMMAND_CLASSES.items():
+            command_arg_parser = subparsers.add_parser(command_name)
+            command_class.add_arguments(command_arg_parser)
+            arg_parsers.append(command_arg_parser)
+​
+        for arg_parser in arg_parsers:
+            arg_parser.add_argument("-c", is_config_file=True, help="config file path")
+            arg_parser.add_argument("--root-path",
+                                           help="path to the mcs-benchmark-data directory")
+            arg_parser.add_argument(
+                "--benchmark-name",
+                help="name of the benchmark the submission was tested against (in snake_case)",
+            )
+            arg_parser.add_argument(

```

Comment:
The benchmark pipeline command doesn't take this, right?. It should be in the classmethod CreateSubmissionPipelineCommand.add_arguments.

Reply:
In any case, probably best for the non-global arguments like benchmark_name and submission_name to be in the commands' add_arguments,  and not here, even if they're reused across commands.